tests/phpunit/tests/rest-api/wpRestBlockPatternCategoriesController.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Unit tests covering WP_Block_Pattern_Categories_Registry functionality.
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ * @since      6.0.0
+ */
+
+/**
+ * Tests for REST API for Block Pattern Categories Registry.
+ *
+ * @since 6.0.0
+ *
+ * @covers WP_REST_Block_Pattern_Categories_Controller
+ *
+ * @group restapi
+ */
+class Tests_REST_WpRestBlockPatternCategoriesController extends WP_Test_REST_Controller_Testcase {
+
+	/**
+	 * Admin user ID.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @var int
+	 */
+	protected static $admin_id;
+
+	/**
+	 * Original instance of WP_Block_Patterns_Registry.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @var WP_Block_Patterns_Registry
+	 */
+	protected static $orig_registry;
+
+	/**
+	 * The REST API route.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @var string
+	 */
+	const REQUEST_ROUTE = '/__experimental/block-patterns/categories';
```

Comment:
The namespace will need to be updated to wp/v2

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/2488/commits/ceedc0a33b0b336e9465bf6322f4464c28f622c8.